packages/EasyPagination/src/Traits/DoctrinePaginatorTrait.php

```diff
@@ -43,7 +43,16 @@ public function getTotalItems(): int
         }
 
         $countAlias = \sprintf('_count%s', $this->fromAlias ? \sprintf('_%s', $this->fromAlias) : '');
-        $queryBuilder = $this->createQueryBuilder()->select(\sprintf('COUNT(1) as %s', $countAlias));
+
+        $queryBuilder = $this->createQueryBuilder()->select(
+            \sprintf(
+                'COUNT(DISTINCT %s) as %s', // Need to distinct to remove duplicates caused by joins.
+                $this->fromAlias,
+                $countAlias
+            )
+        );
+
+        $queryBuilder->resetDQLPart('groupBy'); // Reset groupBy to remove issue when counting.
```

Comment:
What issue is that solving?

Reply:
If a group by exist this will cause the result to have multiple rows.

Reply:
Ok but here we're modifying the SQL query of the app using the package without telling them. What if they actually need the group by?

If it's a real issue to have a groupBy here, I think we should throw an exception to let the app know that they can't add a groupBy in the criteria to count items.

And what happen when we remove the groupBy to count the items but not to fetch them? Doesn't that create inconsistencies as well?

Reply:
Example scenario for when groupBy is needed, is for feature that allows sorting a list of Awards by count of Nominations, so restricting usage of groupBy here will not allow that scenario.

As per testing, items fetched is hydrated, so the itemsPerPage still is per hydrated object count(even with group by). But with totalItems counting, if groupBy exist, this will cause the result to multiple rows and causes inconsistent total count. Removing groupBy + with Count distinct when counting total items seems to be correct way on counting total items.

Haven't tested all scenarios.

Maybe we should move this to a branch, and use it on RR to be safe first?

Reply:
Was thinking this may affect DoctrineDbalLengthAwarePaginator, so maybe move the `getTotalItems` on each subclass instead of in this trait.

DoctrineDbalLengthAwarePaginator contain the old getTotalItems
DoctrineOrmLengthAwarePaginator contain the updated getTotalItems w/c will solve the Alias issue as well.

Reply:
I don't think that this a good approach because the intent was to keep a consistent logic across multiple platforms but now we have 2 different ways of counting items. Have a look at my comment in the issue 😉

Reply:
Yes but DoctrinePaginatorTrait already contains an abstract method: `doGetTotalItems` w/c causes paginators to have separate logic when getting total items.

Added comment on issue: https://github.com/eonx-com/easy-monorepo/issues/297#issuecomment-669573871